File: campaignmanager/src/main/java/com/project/campaignmanager/CampaignService.java
package com.project.campaignmanager;

import jakarta.validation.Valid;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CampaignService {

    private final CampaignRepository campaignRepository;

    public CampaignService(CampaignRepository campaignRepository) {
        this.campaignRepository = campaignRepository;
    }

    public List<Campaign> getAllCampaigns() {
        return campaignRepository.findAll();
    }

    public Optional<Campaign> getCampaignById(Long id) {
        return campaignRepository.findById(id);
    }

    public Campaign createCampaign(Campaign campaign) {
        return campaignRepository.save(campaign);
    }

    public Campaign updateCampaign(Long id, Campaign campaignDetails) {
        return campaignRepository.findById(id).map(campaign -> {
            campaign.setCampaignName(campaignDetails.getCampaignName());
            campaign.setKeywords(campaignDetails.getKeywords());
            campaign.setBidAmount(campaignDetails.getBidAmount());
            campaign.setCampaignFund(campaignDetails.getCampaignFund());
            campaign.setStatus(campaignDetails.getStatus());
            campaign.setTown(campaignDetails.getTown());
            campaign.setRadius(campaignDetails.getRadius());
            return campaignRepository.save(campaign);
        }).orElseThrow(() -> new RuntimeException("Campaign not found"));
    }

    public void deleteCampaign(Long id) {
        campaignRepository.deleteById(id);
    }
}

File: campaign-manager-frontend/src/api/campaignService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/campaigns";

export const getAllCampaigns = () => axios.get(API_URL);
export const getCampaignById = (id) => axios.get(`${API_URL}/${id}`);
export const createCampaign = (campaign) => axios.post(API_URL, campaign);
export const updateCampaign = (id, campaign) => axios.put(`${API_URL}/${id}`, campaign);
export const deleteCampaign = (id) => axios.delete(`${API_URL}/${id}`);
